esp/services/WsDeploy/WsDeployService.cpp

```diff
@@ -4665,7 +4665,13 @@ bool CWsDeployFileInfo::handleEspServiceBindings(IEspContext &context, IEspHandl
         if (resource)
           xpath.appendf("[@resource='%s']", resource);
 
-        IPropertyTree* pSubType = pEnvRoot->queryPropTree(xpath.str());
+        IPropertyTree* pSubType = nullptr;
+        Owned<IPropertyTreeIterator> iterSubType = pEnvRoot->getElements(xpath.str());
+        ForEach(*iterSubType)
```

Comment:
I think the index notation '[1]' might not work due to the attribute filter preceding it [@resource='']. That might be a jlib bug.

Also, we might want to explore making sure the default values for a new row are unique. Other elements tend to append a counter to one of the fields...